tests/conftest.py

```diff
@@ -1902,42 +1903,59 @@ def rgw_bucket_factory_session(request, rgw_obj_session):
 
 
 @pytest.fixture()
-def bucket_factory(request, mcg_obj):
+def bucket_factory(request, bucket_class_factory, mcg_obj):
```

Comment:
there are more than 50 places in which this fixture is being used. Please make sure this change new arg addition is incorporated into these usages

Reply:
This arg should not affect existing calls to `bucket_factory`, since the fixture ultimately returns `_create_buckets()`.
`_create_buckets()` was only changed by the addition of an optional `bucketclass` parameter that defaults to `None`, and thus the change should be invisible to older code.
We also tested all of MCG tier1-3, including tests that were and weren't parametrized, and all of them passed.
I and Elad talked about this in the chat, and so, I'm resolving this conversation.